src/lib/progress-bar/progress-bar.ts

```diff
@@ -113,6 +131,21 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  /** Attaches value bar element animation end listener. */
+  ngAfterViewInit() {
+    if(this.mode === 'determinate' || this.mode === 'buffer') {
```

Comment:
This should also check the `_animationMode`

Reply:
Ok added, to the binding to only bind transition on _animationMode !== noop.